Tests/Invoke-PxGet.Tests.ps1

```diff
@@ -0,0 +1,200 @@
+            Remove-Item -Path "$testRoot \pxget.json"
+        }
+    }
+    
+    function Reset
+    {
+        RemovePxGetFile
+    }
+
+    function ThenErrorThrown
+    {
+        param(
+            [Parameter(Mandatory)]
+            [string] $WithError
+        )
+    
+        $Global:Error[-1] | Should -Match $WithError
+    }
+    
+    function ThenSucceeded
+    {
+        $script:failed | Should -BeFalse
+        $Global:Error | Should -BeNullOrEmpty
+        Assert-MockCalled -CommandName 'Get-RootDirectory' -ModuleName 'PxGet' -Times 2
+    }
+    
+    function WhenInvokingPxGet
+    {
+        try 
+        {
+            Mock -CommandName 'Get-RootDirectory' -ModuleName 'PxGet' { return $testRoot}
+            Invoke-PxGet 'install'
+        }
+        catch 
+        {
+            $script:failed = $true
+            Write-Error -ErrorRecord $_ -ErrorAction $ErrorActionPreference
+        }
+    }
+}
+
+Describe 'Invoke-Pxget' {
+    BeforeEach { 
+        Init
+        New-TestRoot 
+    }
+    AfterEach { 
+        Reset 
+    }
+
+    It 'should pass when there is a valid module in the pxget file' {
+        $contents = @"
+        {
+            "PSModules": [
+                {
+                    "Name": "PackageManagement",
+                    "Version": "1.4.7"
+                }
+            ]
+        }
+"@
+        GivenPxGetFile -Contents $contents
+        WhenInvokingPxGet
+        ThenSucceeded
+    }
+
+    It 'should pass when the module to be installed already exists' {
+        $contents = @"
+        {
+            "PSModules": [
+                {
+                    "Name": "PackageManagement",
+                    "Version": "1.4.7"
+                }
+            ]
+        }
+"@
+        GivenPxGetFile -Contents $contents
+        WhenInvokingPxGet
+        ThenSucceeded
+    }
+
+    It 'should pass when the module to be installed is a prerelease version' {
+        $contents = @"
+        {
+            "PSModules": [
+                {
+                    "Name": "PackageManagement",
+                    "Version": "1.2.0-preview"
+                }
+            ]
+        }
+"@
+        GivenPxGetFile -Contents $contents
+        WhenInvokingPxGet
+        ThenSucceeded
+    }
+
+    It 'should pass when there are multiple modules to be installed' {
+        $contents = @"
+        {
+            "PSModules": [
+                {
+                    "Name": "PackageManagement",
+                    "Version": "1.4.7"
+                },
+                {
+                    "Name": "PowerShellGet",
+                    "Version": "2.2.5"
+                }
+            ]
+        }
+"@
+        GivenPxGetFile -Contents $contents
+        WhenInvokingPxGet
+        ThenSucceeded
+    }
+
+    It 'should run but throw an exception when there is an invalid module name' {
+        $contents = @"
+        {
+            "PSModules": [
+                {
+                    "Name": "Invalid",

```

Comment:
Add another invalid module name and a valid module name (the valid name should be last) and make sure the error message contains both module names and make sure the valid module got installed.